src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,303 @@
+import Box, { BoxProps } from "@vertigis/web/ui/Box";
+import Edit from "@vertigis/web/ui/icons/Edit";
+import { FormElementProps, FormElementRegistration } from "@geocortex/workflow/runtime";
+import IconButton, { IconButtonProps } from "@vertigis/web/ui/IconButton";
+import Input from "@vertigis/web/ui/Input";
+import React, { useState, useEffect, CSSProperties } from "react";
+import Save from "@vertigis/web/ui/icons/Save";
+import Table, { TableProps } from "@vertigis/web/ui/Table";
+import TableBody from "@vertigis/web/ui/TableBody";
+import TableCell, { TableCellProps } from "@vertigis/web/ui/TableCell";
+import TableFooter from "@vertigis/web/ui/TableFooter";
+import TableHead from "@vertigis/web/ui/TableHead";
+import TableRow from "@vertigis/web/ui/TableRow";
+import DynamicIcon from "@vertigis/web/ui/DynamicIcon/DynamicIcon";
+
+type Total = Record<string, any>;
+type Row = Record<string, any>;
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+type SettableTableProps = Pick<TableProps, "size" | "stickyHeader">;
+
+interface Column {
+    alias?: string;
+    align?: TableCellProps["align"];
+    format?: (value: any) => any;
+    parse?: (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => any;
+    includeTotal?: boolean;
+    editable: boolean;
+    type: string;
+    name: string;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<Row[]>,
+        SettableBoxProps,
+        SettableTableProps {
+    cols: Column[];
+    rows: Row[];
+    editTooltip?: string;
+    saveTooltip?: string;
+    includeTotals?: boolean;
+    footerLabel?: string;
+    footerStyle?: CSSProperties;
+    saveIcon?: string;
+    editIcon?: string;
+    onMouseEnter?: (row: Row) => void;
+    onMouseLeave?: (row: Row) => void;
+    onChange?: (data: Row[], row: Row) => void;
+}
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+
+const calculateTotals = (cols: Column[], rows: Row[]): Total => {
```

Comment:
This kind of locks us in to totals only. What about averages and aggregates?

Reply:
I guess we can pass a function for footer calculations.

Reply:
Maybe consider adding something like this to the `Column` interface so that you can easily get a total with baked in logic, but, you can supply your own function, and we can add other well known operations in the future.

`summary?: "sum" | "average" | ((values: any[]) => any)`

Reply:
done.